server/base/src/main/java/org/apache/accumulo/server/conf/CheckCompactionConfig.java

```diff
@@ -0,0 +1,154 @@
+ * under the License.
+ */
+package org.apache.accumulo.server.conf;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+
+import org.apache.accumulo.core.clientImpl.ClientInfoImpl;
+import org.apache.accumulo.core.spi.compaction.DefaultCompactionPlanner;
+import org.apache.accumulo.start.spi.KeywordExecutable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.auto.service.AutoService;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterables;
+import com.google.gson.Gson;
+
+@AutoService(KeywordExecutable.class)
+public class CheckCompactionConfig implements KeywordExecutable {
+
+  private final static Logger log = LoggerFactory.getLogger(CheckCompactionConfig.class);
+
+  public static void main(String[] args) throws IOException {
+    if (args.length != 1)
+      throw new IllegalArgumentException("Only one argument is accepted (path to properties file)");
+
+    Path path = Path.of(args[0]);
+    if (!path.toFile().exists())
+      throw new FileNotFoundException("File at given path could not be found");
+
+    // Extract properties from props file at given path
+    Properties allProps = ClientInfoImpl.toProperties(path);
+    log.info("All props: {}", allProps);
+
+    // Extract server props from set of all props
+    Map<String,String> serverPropsMap = new HashMap<>();
+    String suffix = ".accumulo.server.props";
+    allProps.forEach((k, v) -> {
+      if (k.toString().endsWith(suffix))
+        serverPropsMap.put(k.toString(), v.toString());
+    });
+
+    // Ensure there is exactly one server prop in the map and get its value
+    // The value should be compaction properties
+    String compactionPropertiesString = Iterables.getOnlyElement(serverPropsMap.values());
+
+    // Create a props object for compaction props
+    StringReader sr = new StringReader(compactionPropertiesString.replace(' ', '\n'));
+    Properties serverProps = new Properties();
+    serverProps.load(sr);
+
+    // Extract executors options from compactions props
+    Map<String,String> executorsProperties =
+        getPropertiesWithSuffix(serverProps, ".planner.opts.executors");
+
+    // Ensure there is exactly one executor config in the map and get its value
+    String executorJson = Iterables.getOnlyElement(executorsProperties.values());
+
+    // Convert json string to array of ExecutorConfig objects
+    var executorConfigs =
+        new Gson().fromJson(executorJson, DefaultCompactionPlanner.ExecutorConfig[].class);
+    if (Objects.isNull(executorConfigs)) {
+      throw new RuntimeException("Compaction executors could not be parsed from given file");
+    }
+
+    boolean hasSeenNullMaxSize = false;
+
+    for (var executorConfig : executorConfigs) {
```

Comment:
Suggest re-using the logic at https://github.com/apache/accumulo/blob/main/core/src/main/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlanner.java#L158 so that it's not done in two places.

Reply:
Addressed in 13b39cb